app/src/testWebkit/java/org/mozilla/focus/telemetry/FirebaseEventTest.java

```diff
@@ -0,0 +1,105 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.telemetry;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static junit.framework.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class FirebaseEventTest {
+
+    final private static char FAKE_EVENT_NAME_CHAR = 'a';
+
+    final private static String INVALID_EVENT_NAME;
+    final private static String VALID_EVENT_NAME;
+
+    final private static String INVALID_PARAM_NAME;
+    final private static String VALID_PARAM_NAME;
+
+    final private static String INVALID_PARAM_VALUE;
+    final private static String VALID_PARAM_VALUE;
+
+    final private static int EVENT_NAME_SEPARATOR_LENGTH = FirebaseEvent.EVENT_NAME_SEPARATOR.length() * 2;
+
+    static {
+        // event name already contains two "__", so they are 36 characters left
+        INVALID_EVENT_NAME = prepareString(FirebaseEvent.MAX_LENGTH_EVENT_NAME - EVENT_NAME_SEPARATOR_LENGTH + 1);
+        VALID_EVENT_NAME = prepareString(FirebaseEvent.MAX_LENGTH_EVENT_NAME - EVENT_NAME_SEPARATOR_LENGTH);
+
+        INVALID_PARAM_NAME = prepareString(FirebaseEvent.MAX_LENGTH_PARAM_NAME + 1);
+        VALID_PARAM_NAME = prepareString(FirebaseEvent.MAX_LENGTH_PARAM_NAME);
+
+        INVALID_PARAM_VALUE = prepareString(FirebaseEvent.MAX_LENGTH_PARAM_VALUE + 1);
+        VALID_PARAM_VALUE = prepareString(FirebaseEvent.MAX_LENGTH_PARAM_VALUE);
+
+    }
+
+    private static String prepareString(final int size) {
+        char[] chars = new char[size];
+        Arrays.fill(chars, FAKE_EVENT_NAME_CHAR);
+        return new String(chars);
+    }
+
+    @NonNull
+    private FirebaseEvent generateValidFirebaseEvent() {
+        return FirebaseEvent.create("", "", "", VALID_EVENT_NAME);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void eventNameLongerThan40Characters_shouldThrowIllegalArgumentException() {
+        FirebaseEvent.create("", "", "", INVALID_EVENT_NAME);
+    }
+
+    @Test
+    public void eventNameValid_shouldBeSafe() {
+        try {
+            generateValidFirebaseEvent();
+        } catch (Exception e) {
+            fail("Should not have thrown any exception");
+        }
+    }
+
+
+    @Test
+    public void nullContext_shouldBeSafe() {
+        try {
+            generateValidFirebaseEvent().queue(null);
+        } catch (Exception e) {
+            fail("Should not have thrown any exception");
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void paramSizeTooLarge_shouldThrow() {
+        final FirebaseEvent firebaseEvent = generateValidFirebaseEvent();
+        final Bundle params = mock(Bundle.class);
+        firebaseEvent.setParam(params);
+        when(params.size()).thenReturn(FirebaseEvent.MAX_PARAM_SIZE + 1);
```

Comment:
Let's also fix this to capture the edge case.

Reply:
fixed